server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -197,82 +142,65 @@ private void handleCreate(CreationMessageEntry create, Builder responseBuilder,
             .collect(Guavate.toImmutableList());
     }
 
-    private void performCreate(CreationMessageEntry entry, Builder responseBuilder, MailboxSession session) throws MailboxException, InvalidMailboxForCreationException, MessagingException, AttachmentsNotFoundException {
+    private SetMessagesResponse.Builder performCreate(CreationMessageEntry entry, SetMessagesResponse.Builder builder, MailboxSession session) throws MailboxException, MessagingException {
         if (isAppendToMailboxWithRole(Role.OUTBOX, entry.getValue(), session)) {
```

Comment:
It may be a great improvement to have an enum, for example _CreationKind_ (APPEND_TO_OUTBOX, IS_DRAFT,...)
Then, you just have to call a method which returns the enum for the _CreationMessage_ and switch on it.

No more _if_ _else if_ hell.